src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -4453,7 +4455,7 @@ void G1CollectedHeap::eagerly_reclaim_humongous_regions() {
   G1FreeHumongousRegionClosure cl(&local_cleanup_list);
   heap_region_iterate(&cl);
 
-  remove_from_old_sets(0, cl.humongous_regions_reclaimed());
+  remove_from_old_gen_sets(0, 0, cl.humongous_regions_reclaimed());
```

Comment:
Looking at this call and now having three parameters that are "optional" for `remove_from_old_gen_sets()` I wonder if it would be cleaner to have three functions, one for each set. It would increase the number of times we take the look, but we could restructure the code in `G1ReclaimEmptyRegionsTask` to not do the updates in the worker threads and that way only take the lock when there will be no contention. 

If you fell like this is outside the scope of this change, please file an issue instead.

Reply:
I am not very clear on what the problem there is and how the parameters are optional. I do not completely understand how adding an extra method just for this caller would improve the code significantly.

Reply:
Sounds good.